Eastern-Research-Group/expert-query
# Feature/274 in page navigation

## Related Issues:

- [EQ-268](https://jira.epa.gov/browse/EQ-268)
- [EQ-274](https://jira.epa.gov/browse/EQ-274)
- [EQ-281](https://jira.epa.gov/browse/EQ-281)

## Main Changes:

- Added an in-page navigation component, which allows users to jump to different predetermined locations on the page. This is based on the same [USWDS component](https://designsystem.digital.gov/components/in-page-navigation/), but with custom state management, as the JavaScript side of the USWDS component isn't compatible with React. This has the added benefit of allowing us to style the labels as we please.
- Removed the introduction message box.
- Styled the section headings of the main page as step indicators, and added short descriptions for each. Additionally, the sections are no longer accordions (except for the Advanced Queries section).
- Removed the timeout from the download success/error messages, and implemented logic to dismiss them when there are other state changes.
  - I didn't add the close (x) to the message, because the messages are dismissed easily. **Should this be added?**
- Adjusted some spacing.
- Moved the "Clear Search" button just below the data profile select input. To avoid confusion, I also updated the behavior so that this clears the profile selection, which I think helps clarify that fields were cleared.
- Removed some unnecessary effects.

## Steps To Test:

1. Confirm that the in-page navigation only appears on the main page.
2. Test the behavior of the navigation, and verify that it updates when profiles are switched.
3. Test the download success message. It should dismiss when a new profile is selected, a filter is updated, the file format is changed, or the download button is clicked again.
